multiapps-controller-process/src/test/java/org/cloudfoundry/multiapps/controller/process/steps/CreateOrUpdateAppStepTest.java

```diff
@@ -1,226 +1,180 @@
+    static Stream<Arguments> testHandleApplicationAttributes() {
         return Stream.of(
-// @formatter:off
-            // (0) Disk quota is 0:
-            Arguments.of("create-app-step-input-00.json", null),
-            // (1) Memory is 0:
-            Arguments.of("create-app-step-input-01.json", null),
-            // (2) Everything is specified properly:
-            Arguments.of("create-app-step-input-02.json", null),
-            // (3) Binding parameters exist, and the services do too:
-            Arguments.of("create-app-step-input-03.json", null),
-            // (4) Binding parameters exist, but the services do not:
-            Arguments.of("create-app-step-input-04.json", "Could not bind service \"application\" to application \"service-2\": 500 Internal Server Error: Something happened!"),
-            // (5) Binding parameters exist, but the services do not and service-2 is optional - so no exception should be thrown:
-            Arguments.of("create-app-step-input-05.json", null),
-            // (6) Service keys to inject are specified:
-            Arguments.of("create-app-step-input-06.json", null),
-            // (7) Service keys to inject are specified but not exist:
-            Arguments.of("create-app-step-input-07.json",
-                    "Unable to retrieve required service key element \"expected-service-key\" for service \"existing-service\"")
-// @formatter:on
+//@formatter:off
+                         // (1) Everything is specified properly:
+                         Arguments.of(ImmutableStaging.builder().command("command1").healthCheckType("none").addBuildpack("buildpackUrl").build(),
+                                      128, 256, List.of("example.com", "foo-bar.xyz")),
+                         // (2) Disk quota is 0:
+                         Arguments.of(null, 0, 256, Collections.emptyList()),
+                         // (3) Memory is 0:
+                         Arguments.of(null, 1024, 0, Collections.emptyList())
+//@formatter:on             
         );
     }
 
     @ParameterizedTest
     @MethodSource
-    void testExecute(String stepInput, String expectedExceptionMessage) {
-        initializeParameters(stepInput, expectedExceptionMessage);
-        if (expectedExceptionMessage != null) {
-            assertThrows(SLException.class, () -> step.execute(execution));
-            return;
-        }
+    void testHandleApplicationAttributes(Staging staging, int diskQuota, int memory, List<String> uris) {
+        CloudApplicationExtended application = buildApplication(staging, diskQuota, memory, uris);
+        prepareContext(application, Collections.emptyMap());
+
         step.execute(execution);
-        assertStepFinishedSuccessfully();
 
-        validateClient();
+        assertStepFinishedSuccessfully();
+        verify(client).createApplication(APP_NAME, staging, diskQuota == 0 ? null : diskQuota, memory == 0 ? null : memory, uris, null);
```

Comment:
This line is a bit hard to read. Extract the disk quota and memory into variables